velox/common/memory/MemoryManagerStrategy.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/memory/MemoryManagerStrategy.h"
+#include <unistd.h>
+
+#include <folly/executors/task_queue/UnboundedBlockingQueue.h>
+#include <folly/executors/thread_factory/InitThreadFactory.h>
+
+namespace facebook::velox::memory {
+
+bool MemoryManagerStrategy::initialized_;
+MemoryManagerStrategy::MemoryManagerStrategyFactory
+    MemoryManagerStrategy::factory_;
+
+// static
+MemoryManagerStrategy* MemoryManagerStrategy::instance() {
```

Comment:
We could use system config to configure the strategy to use, and store the strategy object as a static object? Something like:
```
MemoryManagerStrategy* MemoryManagerStrategy::instance() {
  static std::unique_ptr<MemoryManagerStrategy> strategy = createMemoryStrategy(config option);
  return strategy.get();
}

createMemoryStrategy(strategyName) {
   if (factoryMap_.find(strategyName) != factoryMap_.end()) {
      return factoryMap_[strategyName].create();
   } else {
      return createDefault();
   }
}
```